### PR TITLE
Improve gather_nd and scatter_nd doc

### PR DESCRIPTION
**Pull request by haojin2**
## Description ##
Address #11867

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Improve doc for scatter_nd
- [x] Improve doc for gather_nd

## Comments ##
@zheng-da

### CODE REVIEW

File: src/operator/tensor/indexing_op.cc [hunk 1]
```diff
@@ -601,6 +603,17 @@ Examples::
   shape = (2, 2)
   scatter_nd(data, indices, shape) = [[0, 0], [2, 3]]
```
Comment (zheng-da) [inline src/operator/tensor/indexing_op.cc]: the code can actually run?

**Comment by zheng-da** [comment]
could you update [gather_nd](http://mxnet.incubator.apache.org/api/python/symbol/symbol.html#mxnet.symbol.gather_nd), please? they are actually a pair. It's weird to update one, but not the other. Many thanks!

**Comment by haojin2** [comment]
@zheng-da Please also update the issue you raised so we have some references in the future, thanks!

**Comment by zheng-da** [comment]
updated the issue.

**Comment by haojin2** [comment]
@zheng-da Finished addition of extra examples for the slicing use case. Please give a review

**Pull request merged**